### PR TITLE
feat: type_args for create transaction cmd

### PR DESCRIPTION
**Pull request by Rqnsom**
Include type_args for the create-transaction command.
A few quick examples:

```
simple_scripts git:(main) smove create-transaction --compiled-script-path empty_loop_param.mv --type-args u8 u16 u32 u64 u128 u256 bool address 412

ScriptFunctionArguments { type_arg_vec: TypeArgVec { type_args: [U8, U16, U32, U64, U128, U256, Bool, Address, Unparsable("412")] } }

Error: Invalid move type for converting into `TypeTag`:
Unparsable("412")
```

```
simple_scripts git:(main) smove create-transaction --compiled-script-path empty_loop_param.mv --type-args u8 u16 u32 u64 u128 u256 bool address "vector<u16>"

ScriptFunctionArguments { type_arg_vec: TypeArgVec { type_args: [U8, U16, U32, U64, U128, U256, Bool, Address, Vector { items: U16 }] } }

Script transaction is created at:
/home/user/substrate-move/move-vm-backend/tests/assets/move-projects/simple_scripts/build/simple_scripts/script_transactions/empty_loop_param.mvt
```

Note: `type_args.rs` has been inspired by `aptos-core/api/types/src/move_types.rs`

### CODE REVIEW

File: src/cmd/script_args/mod.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use clap::Parser;
+use move_core_types::language_storage::TypeTag;
+use type_args::TypeArgVec;
+
+mod type_args;
+
+#[derive(Debug, Parser)]
+pub struct ScriptFunctionArguments {
+    #[clap(flatten)]
+    pub(crate) type_arg_vec: TypeArgVec,
```
Comment (Rqnsom) [inline src/cmd/script_args/mod.rs]: Docs

Comment (Rqnsom) [reply src/cmd/script_args/mod.rs]: Done

File: src/cmd/script.rs [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+use super::script_transaction::ScriptTransaction;
+use crate::run_context::RunContext;
 use anyhow::{Error, Result};
 use clap::Parser;
 use move_binary_format::file_format::CompiledScript;
 use std::fs;
 use std::path::PathBuf;
 
-use super::script_transaction::ScriptTransaction;
-use crate::run_context::RunContext;
```
Comment (neutrinoks) [inline src/cmd/script.rs:8]: Why did you move the crate-internal imports to the top? I'm asking, because in the past, I usually did the opposite 🙃

File: src/cmd/script_args/mod.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+use type_args::TypeArgVec;
+
+mod type_args;
+
+/// Arguments for script functions.
+#[derive(Debug, Parser)]
+pub struct ScriptFunctionArguments {
+    /// Type args.
+    #[clap(flatten)]
+    pub(crate) type_arg_vec: TypeArgVec,
```
Comment (neutrinoks) [inline src/cmd/script_args/mod.rs]: What is the background for this visibility pattern? Just for my personal interest... :blush: Why not `pub`?

File: src/cmd/script_args/type_args.rs [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+#[derive(Debug, Parser)]
+pub struct TypeArgVec {
+    /// TypeTag arguments separated by spaces.
+    ///
+    /// Example: `u8 u16 u32 u64 u128 u256 bool address vector signer`.
+    #[clap(long, multiple_values(true))]
+    pub(crate) type_args: Vec<MoveType>,
+}
+
+// This function cannot handle the full range of types that MoveType can
```
Comment (neutrinoks) [inline src/cmd/script_args/type_args.rs:16]: Not about the comment, but when reading this, I just wondered, if we have the same editor width (allowed number of characters in one line) in all of our projects :thinking:

File: src/cmd/script_args/type_args.rs [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+// represent. Internally, it uses parse_type_tag, which cannot handle references
+// or generic type parameters. This function adds nominal support for references
+// on top of parse_type_tag, but it still does not work for generic type params.
+// For that, we have the Unparsable variant of MoveType, so the deserialization
+// doesn't fail when dealing with these values.
+impl FromStr for MoveType {
+    type Err = anyhow::Error;
+
+    fn from_str(mut s: &str) -> Result<Self, Self::Err> {
```
Comment (neutrinoks) [inline src/cmd/script_args/type_args.rs]: Why is it `mut s: &str`? :upside_down_face:

**Review by neutrinoks — APPROVED** [review]
LGTM :blush:
My comments are mostly for personal interests only

File: src/cmd/script_args/type_args.rs (hunk 4 above)
Comment (Rqnsom) [reply src/cmd/script_args/type_args.rs:16]: We don't have that yet. We could, but I guess we don't care yet - since it's only a few of us working here - so keeping the code style consistency is easy. But a good point

File: src/cmd/script.rs (hunk 2 above)
Comment (Rqnsom) [reply src/cmd/script.rs:8]: That's not my work. That was done by the all-mighty:
`cargo +nightly fmt --all`

File: src/cmd/script_args/mod.rs (hunk 3 above)
Comment (Rqnsom) [reply src/cmd/script_args/mod.rs]: > What is the background for this visibility pattern?

That's a good question that I'll answer by removing it :D
Since we want to show that this struct element won't be used anywhere else in the codebase, probably ever, and no one should ask the same question again.

File: src/cmd/script_args/type_args.rs (hunk 5 above)
Comment (Rqnsom) [reply src/cmd/script_args/type_args.rs]: _True reason: this code has been purely taken from the `aptos-core/api/types/src/move_types.rs` file._ 🙃
 
But it does seem ugly - let's fix it by doing variable shadowing in the next line. (since we do mutate the `s` in the function)
Done
